python/ray/tune/tests/test_experiment_analysis.py

```diff
@@ -10,10 +10,60 @@
 import pandas as pd
 
 import ray
-from ray.tune import run, sample_from, Analysis
+from ray.tune import run, Trainable, sample_from, Analysis
 from ray.tune.examples.async_hyperband_example import MyTrainableClass
 
 
+class ExperimentAnalysisInMemSuite(unittest.TestCase):
+    scores_list = [5, 4, 2, 9, 7, 0, 1, 8, 6, 3]
+
+    class TrainableScoresList(Trainable):
+        def _setup(self, config):
+            self.timestep = 0
+
+        def _train(self):
+            return {"score": ExperimentAnalysisInMemSuite.scores_list.pop(0)}
+
+        def _save(self, checkpoint_dir):
+            pass
+
+        def _restore(self, checkpoint_path):
+            pass
+
+    def setUp(self):
+        ray.init(local_mode=True, num_cpus=1)
+        self.test_dir = tempfile.mkdtemp()
+        self.test_name = "analysis_exp"
+        self.scores_list_copy = ExperimentAnalysisInMemSuite.scores_list[:]
+        self.run_test_exp()
+
+    def run_test_exp(self):
+        self.ea = run(
+            self.TrainableScoresList,
+            name=self.test_name,
+            local_dir=self.test_dir,
+            stop={"training_iteration": 2},
+            num_samples=5)
```

Comment:
imo just do all this in `testCompareTrials` for modularity, in case we add more tests later